lib/kafo/param.rb

```diff
@@ -40,29 +45,26 @@ def to_s
 
     def set_default(defaults)
       if default == 'UNSET'
-        self.value = nil
+        self.default = nil
       else
         if defaults.has_key?(default)
           value = defaults[default]
           case value
             when :undef
               # value can be set to :undef if value is not defined
               # (e.g. puppetmaster = $::puppetmaster which is not defined yet)
-              self.value = nil
+              self.default = nil
             when :undefined
               # in puppet 2.7 :undefined means that it's param which value is
               # not set by another parameter (e.g. foreman_group = 'something')
               # which means, default is sensible unlike dumped default
               # newer puppet has default dump in format 'value' => 'value' so
               # it's handled correctly by else branch
-              self.value = self.default
             else
-              self.value = value
+              self.default = value
           end
           # if we don't have default value from dump (can happen for modules added from hooks)
           # we fallback to their own default values which must be sensible
-        else
```

Comment:
I suppose this is removed because value loads data from default unless it has been set according to @value_set, we should probably either remove the comment above as well or add one more line explaining that it will be correctly loaded from default value.